CLIKit/Manager.swift

```diff
@@ -9,69 +9,60 @@
 public class Manager {
     public var commands = [Command]()
 
-    public init() {
-    }
-    
+    public init() {}
+
     lazy var defaultCommand: Command = {
         ClosureCommand(name: "", description: "The default command") { argv in
             println("No command specified")
         }
     }()
 
-    public func register(name:String, _ description:String, handler: ClosureCommand.ClosureType) {
+    public func register(name: String, _ description: String, handler: ClosureCommand.ClosureType) {
         register(ClosureCommand(name: name, description: description, handler: handler))
     }
 
-    public func register(command:Command) {
+    public func register(command: Command) {
         commands.append(command)
     }
-    
+
     public func registerDefault(handler: ClosureCommand.ClosureType) {
         defaultCommand = ClosureCommand(name: "", description: "The default command", handler: handler)
     }
 
     /// Finds a command by name
-    public func findCommand(name:String) -> Command? {
-        let foundCommands = commands.filter { $0.name == name }
-        var command:Command?
-
-        if foundCommands.count > 0 {
-            command = foundCommands[0]
-        }
-
-        return command
+    public func findCommand(name: String) -> Command? {
+        return commands.filter { $0.name == name }.first
     }
-    
+
     /// Finds the command to execute based on input arguments
     public func findCommand(argv: ARGV) -> Command? {
         let args = argv.arguments
         // try to find the deepest command name matching the arguments
         for depth in reverse(1...args.count) {
-            let slicedArgs = Array(args[0 ..< depth]) as NSArray
-            let maybeCommandName = slicedArgs.componentsJoinedByString(" ")
-            
+            let slicedArgs = args[0 ..< depth]
+            let maybeCommandName = " ".join(slicedArgs)
+
             if let command = findCommand(maybeCommandName) {
                 argv.arguments = Array(args[depth ..< args.count]) // strip the command name from arguments
                 return command
             }
         }
-        
+
         return nil
     }
-    
+
     /// Runs the correct command based on input arguments
     public func run(arguments: [String]? = nil) {
-        var argv: ARGV!
-        
-        // Swift, Y U NO ||= ?!
-        if arguments != nil {
-            argv = ARGV(arguments!)
+        let argv: ARGV
```

Comment:
Shouldn't this be var (and optional) since you're changing it (and it's currently nil)?

See build failure:

```
⌦  /Users/travis/build/kylef/CLIKit/CLIKit/Manager.swift:56:13: 'let' declarations require an initializer expression
        let argv: ARGV
```